sdk/typescript/src/cryptography/keypair.ts

```diff
@@ -83,11 +86,31 @@ export abstract class BaseSigner {
 	 * The public key for this keypair
 	 */
 	abstract getPublicKey(): PublicKey;
+
+	/**
+	 * The plain bytes for the secret key for this keypair.
+	 */
+	abstract getSecretKeyBytes(): Uint8Array;
 }
 
 /**
- * TODO: Document
+ * This returns an exported keypair object, schema is the signature
+ * scheme name, and the private key field is a Bech32 encoded string
+ * of 33-byte `flag || private_key` that starts with `suiprivkey`.
  */
 export abstract class Keypair extends BaseSigner {
-	abstract export(): ExportedKeypair;
+	export(): ExportedKeypair {
+		const plainBytes = this.getSecretKeyBytes();
+		if (plainBytes.length !== PRIVATE_KEY_SIZE) {
```

Comment:
I'm not sure if this check really belongs here.   I assume we could have some future keypair that doesn't use a key of the same size.   If we want to validate this, I'd rather do it when the class is instantiated, but I would have it in each implementation rather than the base class unless we know t